### DISCUSSION TITLE
How can we integrate a Visual Flow Builder into chaskiq?

### DISCUSSION THREAD

**Discussion by abhishekshank** [chaskiq/chaskiq#627] [Ideas]
Hi

One of the core things which I guess we lack is the Visual Flow Builder in chaskiq.

Can we add formsflow.ai kind of solution into it?

Can you give some idea how to integrate them?

**Comment by michelson** [comment]
Hi @abhishekshank, yes, I'm currently architecting one visual orchestrator of conversational behaviors. It's not released yet, though. The UI thing is not as tricky as the logic that has to be in the backend. For the UI, I'm using https://reactflow.dev/, by the way.

**Comment by abhishekshank** [comment]
Hey Michelson

Great to hear that. How is the output of the designer going to be integrated with the core chat?

I could also add some parts if you can give me little more light.

Regards
Abhishek

**Comment by abhishekshank** [comment]
Hey Michelson

Just to add preciseness to my question, How can I add the csv, json or api exported from the visual designer into the chaskiq core system?

I hope I make sense.

I was just checking botsociety, it allows export directly to dialogflow.

Can we have that kind of connection so that I can integrate an available visual builder directly into chaskiq.

I am sure a lot of paid ones will be willing to provide connect.
